pkg/validation/validate.go

```diff
@@ -86,28 +89,76 @@ func ValidateIAMPolicyResource(ctx context.Context, pDoc v1alpha1.PolicyDocument
 	return nil
 }
 
-//ComparePolicy function compares input policy doc to target policy doc
-func ComparePolicy(ctx context.Context, request string, target string) bool {
+//CompareRole function compares input role to target role
+func CompareRole(ctx context.Context, request awsapi.IAMRoleRequest, role *iam.GetRoleOutput, target string) bool {
```

Comment:
I feel this function is a little bit confusing. CompareRole function compares input role to target role. But, in function parameter `target` (last parameter) is target policy instead of being target role (https://github.com/keikoproj/iam-manager/pull/43/files?file-filters%5B%5D=.go&file-filters%5B%5D=.yaml&file-filters%5B%5D=No+extension#diff-c348e30b9322d2d455da3778316ab4e9R181)

Reply:
So, AWS API doesn't provide permission policy and assume role policy in any of the single api response. Instead, we need to make 2 calls and get those details separately and provide it to compare policy. 
request -> This is what we prepare and does have both input role permission policy and assume role policy and other details too.
role -> This is GetRole response from aws api which has only AssumeRole Policy.
target -> This is Permission Policy from getRolePolicy output.

Let me rename the variables so we don't have much confusion.

Reply:
In other words, Role and RolePolicy are both part of the role comparison as rolePolicy is also the attribute of the wholesome Role.